src/protocol.py

```diff
@@ -202,15 +202,19 @@ def checkIPv6Address(host, hostStandardFormat, private=False):
     Returns hostStandardFormat if it is an IPv6 address,
     otherwise returns False
     """
-    if host == ('\x00' * 15) + '\x01':
+    if host == b'\x00' * 15 + b'\x01':
         if not private:
             logger.debug('Ignoring loopback address: %s', hostStandardFormat)
         return False
-    if host[0] == '\xFE' and (ord(host[1]) & 0xc0) == 0x80:
+    try:
+        host = [ord(c) for c in host[:2]]
+    except TypeError:  # python3 has ints already
+        pass
+    if host[0:1] == b'\xFE' and host[1] & 0xc0 == 0x80:
```

Comment:
I tested it, and the comparison between `int` and `bytes` doesn't work, at least in python2. It should probably be
```
if host[0] == 0xfe and host[1] & 0xc0 == 0x80:
```

Reply:
Oh, I see `host[1]`. In py2 it will be bytes. Perhaps I missed some sample in the `test_checkIPv4Address()`

Reply:
yes, in py2 `host[0]` will be int

Reply:
OK, why don't just write `if host[:2] == b'\xFE\x80'` and what about IPv4 254.128.0.0?

Reply:
And also 169.254.0.0/16

Reply:
Also `test_checkIPv6Address()` should probably fail because the first sample is reserved for documentation:

https://www.ripe.net/participate/member-support/lir-basics/ipv6_reference_card.pdf